occameracontrol/agent.py

```diff
@@ -125,3 +126,23 @@ def next_event(self) -> Event:
         if not events:
             return Event('', 0, 0)
         return events[0]
+
+    def verify_agent(self):
+        '''Verify that an agent exists when it is created
+        '''
+        server = config_rt(str, 'opencast', 'server').rstrip('/')
+        username = config_rt(str, 'opencast', 'username')
+        password = config_rt(str, 'opencast', 'password')
+        auth = (username, password)
+        url = url_parse.quote(
+            f'{server}/capture-admin/agents/{self.agent_id}.json')
```

Comment:
Unfortunately, this leads to a broken URL:

```
✦ ❯ python -m occameracontrol --config uos-camera-control.yml
INFO:root:Updated configuration from uos-camera-control.yml
INFO:root:Log level set to INFO
INFO:occameracontrol.agent:Verification of agent `test`
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/lars/dev/uos/opencast-camera-control/occameracontrol/__main__.py", line 119, in <module>
    main()
  File "/home/lars/dev/uos/opencast-camera-control/occameracontrol/__main__.py", line 89, in main
    agent.verify_agent()
  File "/home/lars/dev/uos/opencast-camera-control/occameracontrol/agent.py", line 141, in verify_agent
    response = requests.get(url, auth=auth, timeout=5)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lars/.local/lib/python3.11/site-packages/requests/api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lars/.local/lib/python3.11/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lars/.local/lib/python3.11/site-packages/requests/sessions.py", line 575, in request
    prep = self.prepare_request(req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lars/.local/lib/python3.11/site-packages/requests/sessions.py", line 486, in prepare_request
    p.prepare(
  File "/home/lars/.local/lib/python3.11/site-packages/requests/models.py", line 368, in prepare
    self.prepare_url(url, params)
  File "/home/lars/.local/lib/python3.11/site-packages/requests/models.py", line 439, in prepare_url
    raise MissingSchema(
requests.exceptions.MissingSchema: Invalid URL 'https%3A//develop.opencast.org/capture-admin/agents/test.json': No scheme supplied. Perhaps you meant https://https%3A//develop.opencast.org/capture-admin/agents/test.json?
```

It also seems like the `requests` library automatically deals with URL encoding:

```
INFO:occameracontrol.agent:Verification of agent `second agent`
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): develop.opencast.org:443
DEBUG:urllib3.connectionpool:https://develop.opencast.org:443 "GET /capture-admin/agents/second%20agent.json HTTP/1.1" 404 42
```

…even with weird Unicode characters in the name:

```
INFO:occameracontrol.agent:Verification of agent `second 🐍 agent`
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): develop.opencast.org:443
DEBUG:urllib3.connectionpool:https://develop.opencast.org:443 "GET /capture-admin/agents/second%20%F0%9F%90%8D%20agent.json HTTP/1.1" 404 42
```

So, let's go back to

```suggestion
        url = f'{server}/capture-admin/agents/{self.agent_id}.json'
```